backend/organization/utility/project_ranking.py

```diff
@@ -57,54 +59,59 @@ def calculate_ranking(
+            last_project_follower = (
+                ProjectFollower.objects.filter(project_id=project_id)
+                .order_by("-created_at")
+                .last()
+            )
+            last_project_follower_timestamp = None if not last_project_follower else last_project_follower.created_at.timestamp()
 
-        project_factors = {
-            "total_comments": ProjectComment.objects.filter(project_id=project_id).count(),
-            "total_likes": ProjectLike.objects.filter(project_id=project_id).count(),
-            "total_followers": ProjectFollower.objects.filter(project_id=project_id).count(),
-            "last_project_comment": self.calculate_recency_of_interaction(
-                last_interaction_timestamp=last_project_comment_timestamp
-            ),
-            "last_project_like": self.calculate_recency_of_interaction(
-                last_interaction_timestamp=last_project_like_timestamp
-            ),
-            "last_project_follower": self.calculate_recency_of_interaction(
-                last_interaction_timestamp=last_project_follower_timestamp
-            ),
-            "total_tags": ProjectTagging.objects.filter(project_id=project_id).count(),
-            "location": 1 if location else 0,
-            "description": 1 if description and len(description) > 0 else 0,
-            "total_skills": total_skills
-        }
+            project_factors = {
+                "total_comments": ProjectComment.objects.filter(project_id=project_id).count(),
+                "total_likes": ProjectLike.objects.filter(project_id=project_id).count(),
+                "total_followers": ProjectFollower.objects.filter(project_id=project_id).count(),
+                "last_project_comment": self.calculate_recency_of_interaction(
+                    last_interaction_timestamp=last_project_comment_timestamp
+                ),
+                "last_project_like": self.calculate_recency_of_interaction(
+                    last_interaction_timestamp=last_project_like_timestamp
+                ),
+                "last_project_follower": self.calculate_recency_of_interaction(
+                    last_interaction_timestamp=last_project_follower_timestamp
+                ),
+                "total_tags": ProjectTagging.objects.filter(project_id=project_id).count(),
+                "location": 1 if location else 0,
+                "description": 1 if description and len(description) > 0 else 0,
+                "total_skills": total_skills
+            }
+
+            project_rank = int(sum(project_factors[factor] * weights[factor] for factor in weights))
+            cache.set(cache_key, project_rank)
 
-        project_rank = int(sum(project_factors[factor] * weights[factor] for factor in weights))
         # Now we want to change the order every time somebody
         # visits the page we would add some randomization.
-        project_rank /= self._randomize()
+        project_rank = self._randomize()
```

Comment:
How much randomization does this do? It seems completely random now to me. I feel like it should be a hard factor and the randomization should do something between decreasing by 20% and increasing by 20%. We'd have to do some trial and error to find the sweet spot here. Otherwise it looks great, we just need to do some fine-tuning on the factors in user-testing with Thomas and Tobi :)

Reply:
This is fixed.